script/deploy.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+set -e
+
+contract_name="$1"
+provided_chain_name="$2"
+
+if [ -z "$contract_name" ]; then
+  echo "Usage: ./script/deploy.sh <contract_name (rln)> <chain_name (sepolia, polygon-zkevm-testnet)>"
+  exit 1
+fi
+
+if [ -z "$provided_chain_name" ]; then
+  echo "Usage: ./script/deploy.sh <contract_name (rln)> <chain_name (sepolia, polygon-zkevm-testnet)>"
+  exit 1
+fi
+
+echo "Sourcing .env"
+source .env
+
+rpc_url=""
+# Check if appropriate env vars are set
+if [ "$provided_chain_name" = "sepolia" ]; then
+  if [ -z "$SEPOLIA_RPC_URL" ]; then
+    echo "SEPOLIA_RPC_URL is not set"
+    exit 1
+  else
+    rpc_url="$SEPOLIA_RPC_URL"
+  fi
+elif [ "$provided_chain_name" = "polygon-zkevm-testnet" ]; then
+  if [ -z "$POLYGON_ZKEVM_TESTNET_RPC_URL" ]; then
+    echo "POLYGON_ZKEVM_TESTNET_RPC_URL is not set"
+    exit 1
+  else
+    rpc_url="$POLYGON_ZKEVM_TESTNET_RPC_URL"
+  fi
+else
+  echo "Invalid chain name, try again with sepolia/polygon-zkevm-testnet"
+  exit 1
+fi
+
+if [ -z "$PRIVATE_KEY" ]; then
+  echo "PRIVATE_KEY is not set"
+  exit 1
+fi
+
+echo "Deploying $contract_name..."
+
+# Deploy the contract
+if [ "$contract_name" = "rln" ]; then
+  chain_id=$(cast chain-id --rpc-url "$rpc_url")
+
+  chain_name=""
+  verifier_url=""
+  if [ -z "$chain_id" ]; then
+    echo "Failed to get chain id"
+    exit 1
+  fi
+
+  if [ "$chain_id" = "11155111" ]; then
+    chain_name="sepolia"
+  elif [ "$chain_id" = "1442" ]; then
+    chain_name="polygon-zkevm-testnet"
+  else
+    echo "Invalid chain id, try again with sepolia/polygon-zkevm-testnet"
+    exit 1
+  fi
+
+
+  forge script script/Deploy.s.sol:Deploy --chain $chain_name --rpc-url $rpc_url --private-key "$PRIVATE_KEY" --broadcast -v
```

Comment:
FYI: Since we're using `Deploy.s.sol` here, which is `BaseScript`, we can probably better use the `MNEMONIC` environment variable over `--private-key`.

Otherwise, I believe there's also an interactive mode in `forge script` using `-i` which lets you enter the private key. Not sure if this is any better though.